zaza/openstack/utilities/openstack.py

```diff
@@ -181,18 +183,70 @@
              'ceilometer and gnocchi')}}
 
 # For vault TLS certificates
+CACERT_FILENAME_FORMAT = "{}_juju_ca_cert.crt"
+CERT_PROVIDERS = ['vault']
+LOCAL_CERT_DIR = "tests"
+REMOTE_CERT_DIR = "/usr/local/share/ca-certificates"
 KEYSTONE_CACERT = "keystone_juju_ca_cert.crt"
 KEYSTONE_REMOTE_CACERT = (
     "/usr/local/share/ca-certificates/{}".format(KEYSTONE_CACERT))
-KEYSTONE_LOCAL_CACERT = ("tests/{}".format(KEYSTONE_CACERT))
+KEYSTONE_LOCAL_CACERT = ("{}/{}".format(LOCAL_CERT_DIR, KEYSTONE_CACERT))
+
+
+async def async_block_until_ca_exists(application_name, ca_cert,
+                                      model_name=None, timeout=2700):
+    """Block until a CA cert is on all units of application_name.
+
+    :param application_name: Name of application to check
+    :type application_name: str
+    :param ca_cert: The certificate to look for.
+    :type ca_cert: str
+    :param model_name: Name of model to query.
+    :type model_name: str
+    :param timeout: How long in seconds to wait
+    :type timeout: int
+    """
+    async def _check_ca_present(model, ca_files):
+        units = model.applications[application_name].units
+        for ca_file in ca_files:
+            for unit in units:
+                try:
+                    output = await unit.run('cat {}'.format(ca_file))
+                    contents = output.data.get('results').get('Stdout', '')
+                    if ca_cert not in contents:
+                        break
+                # libjuju throws a generic error for connection failure. So we
+                # cannot differentiate between a connectivity issue and a
+                # target file not existing error. For now just assume the
+                # latter.
+                except JujuError:
+                    continue
```

Comment:
This should be `break`, as `continue` stands for "we have found the cert", right?

Reply:
Good spot, thanks!